dbcon/execplan/treenode.h

```diff
@@ -1031,41 +1034,75 @@ inline IDB_Decimal TreeNode::getDecimalVal()
             throw logging::InvalidConversionExcept("TreeNode::getDecimalVal: non-support conversion from binary string");
 
         case CalpontSystemCatalog::BIGINT:
+        {
+            int64_t temp = (int64_t)(fResult.intVal * pow((double)10.0, fResultType.scale));
+
+            if (fResultType.isWideDecimalPrecision())
+                fResult.decimalVal = IDB_Decimal(0, fResultType.scale,
+                                         fResultType.precision, (int128_t)temp);
+            else
+                fResult.decimalVal = IDB_Decimal(temp, fResultType.scale,
+                                         fResultType.precision);
+
+            break;
+        }
+
         case CalpontSystemCatalog::MEDINT:
         case CalpontSystemCatalog::INT:
         case CalpontSystemCatalog::SMALLINT:
         case CalpontSystemCatalog::TINYINT:
-            fResult.decimalVal.value = (int64_t)(fResult.intVal * pow((double)10.0, fResultType.scale));
-            fResult.decimalVal.scale = fResultType.scale;
-            fResult.decimalVal.precision = fResultType.precision;
+            fResult.decimalVal = IDB_Decimal(
+                                     (int64_t)(fResult.intVal * pow((double)10.0, fResultType.scale)),
+                                     fResultType.scale, fResultType.precision);
```

Comment:
It seems fResultType.scale should always be 0 here. I think we can remove the pow() part and add this:

`idbassert(fResultType.scale == 0);`

Reply:
I agree, `fResultType.scale` here should be 0. Updated the patch.